crates/hir-expand/src/builtin_fn_macro.rs

```diff
@@ -396,21 +396,19 @@ fn use_panic_2021(db: &dyn ExpandDatabase, span: Span) -> bool {
     // To determine the edition, we check the first span up the expansion
     // stack that does not have #[allow_internal_unstable(edition_panic)].
     // (To avoid using the edition of e.g. the assert!() or debug_assert!() definition.)
-    loop {
-        let Some(expn) = db.lookup_intern_syntax_context(span.ctx).outer_expn else {
-            break false;
-        };
-        let expn = db.lookup_intern_macro_call(expn);
-        // FIXME: Record allow_internal_unstable in the macro def (not been done yet because it
-        // would consume quite a bit extra memory for all call locs...)
-        // if let Some(features) = expn.def.allow_internal_unstable {
-        //     if features.iter().any(|&f| f == sym::edition_panic) {
-        //         span = expn.call_site;
-        //         continue;
-        //     }
-        // }
-        break expn.def.edition >= Edition::Edition2021;
-    }
+    let Some(expn) = db.lookup_intern_syntax_context(span.ctx).outer_expn else {
+        return false;
+    };
+    let expn = db.lookup_intern_macro_call(expn);
+    // FIXME: Record allow_internal_unstable in the macro def (not been done yet because it
+    // would consume quite a bit extra memory for all call locs...)
+    // if let Some(features) = expn.def.allow_internal_unstable {
+    //     if features.iter().any(|&f| f == sym::edition_panic) {
+    //         span = expn.call_site;
+    //         continue;
+    //     }
+    // }
+    expn.def.edition >= Edition::Edition2021
```

Comment:
Please undo this again, this is a loop because of the commented out code and so has meaning once thats fixed again

Reply:
Sure

Reply:
I've reset the branch to not include the manual changes. They will come in a separate PR.